src/Lecture2.hs

```diff
@@ -178,7 +269,14 @@ False
 True
 -}
 isIncreasing :: [Int] -> Bool
-isIncreasing = error "TODO"
+isIncreasing = go True
+  where
+    go :: Bool -> [Int] -> Bool
+    go acc [] = acc
+    go acc [_] = acc
+    go acc (x : xs)
+      | x < head xs = go acc xs
+      | otherwise = False
```

Comment:
You never update the value of `acc` variable and it will be `True` in the end as the initial value. That means that you can safely remove this variable.

Also, you can implement this function even more elegantly if you use nested pattern matching on lists. There will be no need to use unsafe functions like `head` 🙂